pkgs/tools/networking/unbound/default.nix

```diff
@@ -110,7 +110,7 @@ stdenv.mkDerivation rec {
 
   checkInputs = [ bison ];
 
-  doCheck = true;
+  doCheck = stdenv.hostPlatform == stdenv.buildPlatform;
```

Comment:
why is this cross commit in this PR?

these changes also aren't necessary https://github.com/NixOS/nixpkgs/blob/c4961615c7b65e4186076ae3bf521eb40ad48f91/pkgs/stdenv/generic/make-derivation.nix#L168

Reply:
The snippet above is broken. I needed to add this because unbound wouldn't build.

Reply:
Try `with pkgsCross.armv7l-hf-multiplatform.stdenv; buildPlatform.canExecute hostPlatform` in a repl.

Reply:
(The reason it's here is because I wanted to validate cross still works.)

Reply:
`pkgsCross.aarch64-multiplatform.unbound` and so does `pkgsCross.armv7l-hf-multiplatform.unbound` from x86_64-linux

Reply:
`nix-build -A pkgsCross.armv7l-hf-multiplatform.unbound` after disabling doCheck on nettle since that's now broken too:

    error: builder for '/nix/store/fvxq7hkrwj05k0dg7jl5n24fqjm9p7mq-unbound-armv7l-unknown-linux-gnueabihf-1.17.0.drv' failed with exit code 2;
           last 10 log lines:
           > libtool: compile:  armv7l-unknown-linux-gnueabihf-gcc -I. -I/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/include -I/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/include -I/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/include -DSRCDIR=. -g -O2 -flto -c testcode/testbound.c  -fPIC -DPIC -o .libs/testbound.o
           > ./libtool --tag=CC --mode=compile armv7l-unknown-linux-gnueabihf-gcc -I.  -I/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/include -I/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/include -I/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/include -DSRCDIR=. -g -O2 -flto -fPIE  -o replay.lo -c testcode/replay.c
           > libtool: compile:  armv7l-unknown-linux-gnueabihf-gcc -I. -I/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/include -I/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/include -I/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/include -DSRCDIR=. -g -O2 -flto -c testcode/replay.c  -fPIC -DPIC -o .libs/replay.o
           > ./libtool --tag=CC --mode=compile armv7l-unknown-linux-gnueabihf-gcc -I.  -I/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/include -I/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/include -I/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/include -DSRCDIR=. -g -O2 -flto -fPIE  -o fake_event.lo -c testcode/fake_event.c
           > libtool: compile:  armv7l-unknown-linux-gnueabihf-gcc -I. -I/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/include -I/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/include -I/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/include -DSRCDIR=. -g -O2 -flto -c testcode/fake_event.c  -fPIC -DPIC -o .libs/fake_event.o
           > ./libtool --tag=CC --mode=link armv7l-unknown-linux-gnueabihf-gcc  -R/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/lib -R/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/lib -I.  -I/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/include -I/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/include -I/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/include -DSRCDIR=. -g -O2 -flto -fPIE -pie -Wl,-z,relro,-z,now -L/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/lib -L/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/lib -L/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/lib -o testbound testbound.lo replay.lo fake_event.lo testpkts.lo worker.lo acl_list.lo daemon.lo stats.lo shm_main.lo dns.lo infra.lo rrset.lo dname.lo msgencode.lo as112.lo msgparse.lo msgreply.lo packed_rrset.lo iterator.lo iter_delegpt.lo iter_donotq.lo iter_fwd.lo iter_hints.lo iter_priv.lo iter_resptype.lo iter_scrub.lo iter_utils.lo localzone.lo mesh.lo modstack.lo view.lo outbound_list.lo alloc.lo config_file.lo configlexer.lo configparser.lo fptr_wlist.lo edns.lo locks.lo log.lo mini_event.lo module.lo net_help.lo random.lo rbtree.lo regional.lo rtt.lo dnstree.lo lookup3.lo lruhash.lo slabhash.lo tcp_conn_limit.lo timehist.lo tube.lo winsock_event.lo autotrust.lo val_anchor.lo rpz.lo proxy_protocol.lo validator.lo val_kcache.lo val_kentry.lo val_neg.lo val_nsec3.lo val_nsec.lo val_secalgo.lo val_sigcrypt.lo val_utils.lo dns64.lo  authzone.lo         respip.lo ub_event.lo keyraw.lo sbuffer.lo wire2str.lo parse.lo parseutil.lo rrdef.lo str2wire.lo memcmp.lo malloc.lo strlcat.lo strlcpy.lo arc4random.lo arc4random_uniform.lo arc4_lock.lo -lssl -levent  -lcrypto
           > libtool: link: armv7l-unknown-linux-gnueabihf-gcc -I. -I/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/include -I/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/include -I/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/include -DSRCDIR=. -g -O2 -flto -fPIE -pie -Wl,-z -Wl,relro -Wl,-z -Wl,now -o testbound .libs/testbound.o .libs/replay.o .libs/fake_event.o .libs/testpkts.o .libs/worker.o .libs/acl_list.o .libs/daemon.o .libs/stats.o .libs/shm_main.o .libs/dns.o .libs/infra.o .libs/rrset.o .libs/dname.o .libs/msgencode.o .libs/as112.o .libs/msgparse.o .libs/msgreply.o .libs/packed_rrset.o .libs/iterator.o .libs/iter_delegpt.o .libs/iter_donotq.o .libs/iter_fwd.o .libs/iter_hints.o .libs/iter_priv.o .libs/iter_resptype.o .libs/iter_scrub.o .libs/iter_utils.o .libs/localzone.o .libs/mesh.o .libs/modstack.o .libs/view.o .libs/outbound_list.o .libs/alloc.o .libs/config_file.o .libs/configlexer.o .libs/configparser.o .libs/fptr_wlist.o .libs/edns.o .libs/locks.o .libs/log.o .libs/mini_event.o .libs/module.o .libs/net_help.o .libs/random.o .libs/rbtree.o .libs/regional.o .libs/rtt.o .libs/dnstree.o .libs/lookup3.o .libs/lruhash.o .libs/slabhash.o .libs/tcp_conn_limit.o .libs/timehist.o .libs/tube.o .libs/winsock_event.o .libs/autotrust.o .libs/val_anchor.o .libs/rpz.o .libs/proxy_protocol.o .libs/validator.o .libs/val_kcache.o .libs/val_kentry.o .libs/val_neg.o .libs/val_nsec3.o .libs/val_nsec.o .libs/val_secalgo.o .libs/val_sigcrypt.o .libs/val_utils.o .libs/dns64.o .libs/authzone.o .libs/respip.o .libs/ub_event.o .libs/keyraw.o .libs/sbuffer.o .libs/wire2str.o .libs/parse.o .libs/parseutil.o .libs/rrdef.o .libs/str2wire.o .libs/memcmp.o .libs/malloc.o .libs/strlcat.o .libs/strlcpy.o .libs/arc4random.o .libs/arc4random_uniform.o .libs/arc4_lock.o  -L/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/lib -L/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/lib -L/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/lib -lssl /nix/store/w771s40v542jmi4hyq2l3v0y96xxxia7-libevent-armv7l-unknown-linux-gnueabihf-2.1.12/lib/libevent.so -lcrypto -Wl,-rpath -Wl,/nix/store/w771s40v542jmi4hyq2l3v0y96xxxia7-libevent-armv7l-unknown-linux-gnueabihf-2.1.12/lib -Wl,-rpath -Wl,/nix/store/w771s40v542jmi4hyq2l3v0y96xxxia7-libevent-armv7l-unknown-linux-gnueabihf-2.1.12/lib -Wl,-rpath -Wl,/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/lib -Wl,-rpath -Wl,/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/lib
           > ./unittest
           > /nix/store/sc8jyzxrzkw4sq01fa9zbgj530prz2sy-bash-5.2-p15/bin/bash: line 1: ./unittest: cannot execute binary file: Exec format error
           > make: *** [Makefile:345: test] Error 126
           For full logs, run 'nix log /nix/store/fvxq7hkrwj05k0dg7jl5n24fqjm9p7mq-unbound-armv7l-unknown-linux-gnueabihf-1.17.0.drv'.

Reply:
Took the cross fixes out; that can come at a later day.